cfme/tests/infrastructure/test_host.py

```diff
@@ -1,30 +1,45 @@
 # -*- coding: utf-8 -*-
 import pytest
 import random
+import socket
 
 from cfme.base.credential import Credential
 from cfme.common.host_views import HostsEditView
 from cfme.common.provider_views import ProviderNodesView
 from cfme.infrastructure.provider import InfraProvider
 from cfme.infrastructure.provider.rhevm import RHEVMProvider
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
+from cfme.utils.appliance.implementations.ui import navigate_to
 from cfme.utils.blockers import BZ
 from cfme.utils.conf import credentials
-from cfme.utils.appliance.implementations.ui import navigate_to
-from cfme.utils.providers import ProviderFilter
-from markers.env_markers.provider import providers
-
-
-all_infra_prov = ProviderFilter(classes=[InfraProvider], required_fields=['hosts'])
+from cfme.utils.wait import wait_for
 
 pytestmark = [
     pytest.mark.tier(3),
-    pytest.mark.provider(gen_func=providers,
-                         filters=[all_infra_prov,
-                                  lambda provider: len(provider.data.get('hosts', {})) > 1],
-                         scope='module'),
+    pytest.mark.provider([InfraProvider], required_fields=['hosts'], scope='module'),
 ]
 
 
+@pytest.fixture(scope='module')
+def hosts_ipaddresses(provider):
+    ipaddresses = []
+    all_hosts = provider.data.get('hosts', [])
+    for host in all_hosts:
+        if hasattr(host, 'ipaddress'):
+            ipaddr = host.ipaddress
+        else:
+            try:
+                ipaddr = socket.gethostbyname(host.name)
+            except Exception:
```

Comment:
Possibly ``socket.error``, ``socket.herror``, ``socket.timeout`` and ``NameError`` can all be raised here. Since I want to catch everything that can go wrong here anyway, what's the point in listing all the exceptions?